cmd/kubetoken/kubetoken.go

```diff
@@ -54,25 +54,36 @@ You can find more information on 'ExecCredential' and 'client side authenticatio
 	cmd.RunE = func(_ *cobra.Command, args []string) error {
 		ctx := context.Background()
 
-		err := contextCMD.NewContextCommand().Run(ctx, &contextCMD.ContextOptions{
+		k8sClientProvider := okteto.NewK8sClientProvider()
+		okClientProvider := okteto.NewOktetoClientProvider()
+		err := newPreReqValidator(
+			withCtxName(contextName),
+			withNamespace(namespace),
+			withK8sClientProvider(k8sClientProvider),
+			withOktetoClientProvider(okClientProvider),
+		).Validate(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to validate pre-reqs: %w", err)
+		}
+
+		err = contextCMD.NewContextCommand().Run(ctx, &contextCMD.ContextOptions{
 			Context:   contextName,
 			Namespace: namespace,
 		})
 		if err != nil {
 			return err
 		}
 
-		octx := okteto.Context()
-		if !octx.IsOkteto {
-			return errors.ErrContextIsNotOktetoCluster
+		if err := newK8sSupportValidator(k8sClientProvider).Validate(ctx); err != nil {
```

Comment:
We need to download the k8s cfg from the context initialization. If we end up not having this validation it will be easier

Reply:
could be added in parallel with others like ns validation or okteto support validation?

Reply:
We need to get the k8s config to connect to the cluster. This is done at the context init step. My first though was to do it also all together to reduce the time but I can't do it with this one